zha/application/platforms/sensor/const.py

```diff
@@ -390,3 +390,36 @@ class SensorDeviceClass(enum.StrEnum):
     SensorDeviceClass.ENUM,
     SensorDeviceClass.TIMESTAMP,
 }
+
+
+class AnalogInputStateClass(enum.Enum):
+    """State class for AnalogInput Types."""
+
+    TEMPERATURE = (0x00, SensorDeviceClass.TEMPERATURE)
```

Comment:
Let's move this enum to zigpy. We generally don't want to define ZCL enums and types in the ZHA library itself. From the spec:

<img width="761" alt="image" src="https://github.com/user-attachments/assets/a32df1d0-2975-4263-9b05-f4fcf85f4f16">

I think we can define this easily in zigpy:

```python
class AnalogInputType(t.enum8):
    Temp_In_Degrees_C = 0x00
    ...


ANALOG_INPUT_TYPES = {
    AnalogInputType.Temp_In_Degrees_C: TempInDegreesC,
}


class TempInDegreesC(t.enum16):
    Two_Pipe_Entering_Water_Temperature = 0x0000
    ...


class ApplicationType(t.uint32_t, t.Struct):
    group: t.uint8_t
    type: AnalogInputType
    index: t.uint16_t
```

And then ZHA will only deal with creating a mapping between the ZCL spec units and the HA device classes.